### ISSUE TITLE
Console messages don't display, even though X-ChromeLogger-Data header exists

### ISSUE THREAD

**Issue by stillnet** [ccampbell/chromelogger#60]
I'm trying to use chromelogger in an SPA. I make ajax calls, and on the server side I use chromelogger on that request. When the request completes I can see a lot of data in the X-ChromeLogger-Data header, but I don't get data in my console. *occasionally* the data shows up, but most of the time it doesn't.

Any ideas? The chromelogger icon in chrome is enabled. I'm on chrome 42.0.2311.13. Just installed the chromelogger extension yesterday so I'm on the latest version.

Thanks.

**Comment by ccampbell** [comment]
Have you clicked the icon to make sure it is toggled on?

![](https://craig.global.ssl.fastly.net/img/chromelogger/toggle.gif)

**Comment by stillnet** [comment]
Yes as I said the chromelogger icon is enabled. I've togged it off and back on to see if that made a difference and it didn't.

**Comment by ccampbell** [comment]
Ah sorry about that... hmmm... Can you try it on a regular request that is not ajax?  

If you go to https://craig.is/writing/chrome-logger and open the console and toggle on the extension do you see the logs?

**Comment by ccampbell** [comment]
Interesting... I am actually not seeing them show up every time too.  I wonder if this is something specific to a new version of chrome.  I will have to dig into it more later.

**Comment by stillnet** [comment]
Thanks. I noticed the irregularity when testing on a single page also.

**Comment by filso** [comment]
Hi, I believe there's some race condition there. maybe `_listenForLogMessages` runs too late, after response is back?

**Comment by veryxcit** [comment]
Doesn't work reliably I'm afraid. Probably because Chrome does goofy stuff to the headers/caches them? Its though to use in production.

**Comment by ccampbell** [comment]
Is there a page where this is consistently reproducible or some other way I can test it?

**Comment by DeadpoolKerns** [comment]
I am not sure if you've identified any oddities with different Chrome releases but I'm experiencing similar issues. I see the X-ChromeLogger-Data but I don't see the message sent from my PHP script appear in the console.  I've traced the session with Chrome developer tools, decoded the logger-data, and see the log message from the script.  See attached script and Json encoded trace of HTTP request/response. Let me know if you find anything - Thanks.
[PHPConsoleTest.zip](https://github.com/ccampbell/chromelogger/files/1254795/PHPConsoleTest.zip)

**Comment by DeadpoolKerns** [comment]
Well - just figured out the issue.  In case anyone else has this issue - my issue was as simple as not selecting the Info type errors in the console. For anyone else experiencing this issue in getting messages in Chrome developer console make sure the error types you want to see are selected.
![phpconsole](https://user-images.githubusercontent.com/29611751/29755939-4057a0d6-8b6d-11e7-9055-eb1c5f0025a7.png)
